src/utils/retries.ts

```diff
@@ -0,0 +1,87 @@
+import { PineconeMaxRetriesExceededError } from '../errors';
+
+/** Configuration options for retry logic */
+export interface RetryOptions {
+  maxRetries: number;
+  delayStrategy?: string;
+}
+
+/* Retry asynchronous operations.
+ *
+ * @param asyncFn - The asynchronous function to retry.
+ * @param options - The configuration options for retry logic.
+ * */
+export class RetryOnServerFailure {
+  maxRetries: number;
+  delayStrategy: string;
+  asyncFn: (...args: any[]) => Promise<any>;
+
+  constructor(
+    asyncFn: (...args: any[]) => Promise<any>,
+    options?: RetryOptions
+  ) {
+    if (options) {
+      this.maxRetries = options.maxRetries;
+      if (options.delayStrategy) {
+        this.delayStrategy = options.delayStrategy;
```

Comment:
We might want to remove this (lmk what you think, reviewers), since it might be confusing to end users -- basically if they put in any string other than `jitter`, it'll just default to a 1s delay (as opposed to doing something unique for the string `exponential` or something).